Add --rm flag for docker commands and remove useless pull step

NicolasCARPi opened the pull request:
Docker run commands need to have the `--rm` flag to avoid stopped containers piling up after several use.

Also, Docker will automatically pull the image when the run command is launched, so there is no need for this extra step of prealably pulling the image.

a-kuznetsova commented:
Thank you for your suggestions. You are right, Docker will automatically pull the image when the run command is launched, but this will only happen during the first run. Since we want our users to have the latest version possible, we included the `docker pull` command to the Readme. Could you please mention this in the Readme next to the `docker pull` command?

Regarding the `--rm` flag, let's add it as you suggest :)

NicolasCARPi commented:
@a-kuznetsova You're correct, the pull command is nice to have to get the latest version. I added `--rm` to CI commands too even if it's not strictly necessary as the whole thing has a short lifespan, but it's to be consistent. I removed the `-it` as it's not necessary. Also removed trailing whitespaces and added newline at end of file. And added the `--rm` on main README.

a-kuznetsova commented on Docker/README.md:
````diff
@@ -87,15 +93,15 @@ With the additional parameter provided, the resulting command will look as follo
 
 - For local execution with the results in the UI:
      ```
-        docker run -it -p 8080:8080 \
````
@sepich please review this line.

sepich replied:
This command (`--show-report`) runs container which would never exit. 
We need `-it` here to be able to stop serving of UI via `Ctrl-C`

NicolasCARPi replied:
even with `-it`, `Ctrl-c` doesn't work for me. It does for you?

sepich replied:
Yes, but it is broken in this exact container version. It will work in next one.

NicolasCARPi replied:
added back the -it then.

sepich reviewed (approved):
LGTM
